python/cudf/cudf/pylibcudf_tests/test_avro.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import io
+import itertools
+
+import fastavro
+import pyarrow as pa
+import pytest
+from utils import assert_table_and_meta_eq
+
+import cudf._lib.pylibcudf as plc
+
+avro_dtype_pairs = [
+    ("boolean", pa.bool_()),
+    ("int", pa.int32()),
+    ("long", pa.int64()),
+    ("float", pa.float32()),
+    ("double", pa.float64()),
+    ("bytes", pa.string()),
+    ("string", pa.string()),
+]
+
+
+@pytest.fixture(
+    scope="module",
+    params=itertools.product(avro_dtype_pairs, avro_dtype_pairs),
```

Comment:
*question*: Do we need the full cartesian product, or, since the matrix is symmetric, only the lower-triangular part?

Reply:
Good point.

I changed to itertools.combinations, which I think should do what we want.